src/components/SummaryBar/__tests__/SummaryBar.test.js

```diff
@@ -0,0 +1,176 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import { Provider } from 'react-redux';
+import configureMockStore from 'redux-mock-store';
+import SummaryBar from '../SummaryBar';
+import MockAdapter from 'axios-mock-adapter';
+import axios from 'axios'; 
+
+const mockStore = configureMockStore();
+const store = mockStore({});
+
+
+const mock = new MockAdapter(axios);
+describe('SummaryBar Component', () => {
+  let wrapper;
+
+  const props = {
+    displayUserId: '123',
+    summaryBarData: { tangibletime: 10 },    
+    auth: {
+      user: {
+        userid: '123', // Assuming 'authUser' expects a 'userid' field
+      }
+    },
+    userProfile: {
+      _id: '123',
+      firstName: 'John',
+      lastName: 'Doe',
+      email: 'john@example.com',
+      role: 'Owner',
+      weeklycommittedHours: 40,
+      missedHours: 5, 
+      infringements: [], 
+      badgeCollection: [], 
+    },
+    userTask: [],
+    hasPermission: jest.fn(),
+    toggleSubmitForm: jest.fn(),
+    submittedSummary: false,
+
+  };
+
+  beforeEach(() => {
+    wrapper = shallow(
+      <Provider store={store}>
+        <SummaryBar {...props} />
+      </Provider>
+    );
+  });
+
+  it('should render without crashing', () => {
+    expect(wrapper.exists()).toBe(true);
+  });
+
+  it('should display loading when displayUserProfile and summaryBarData are undefined', () => {
+    wrapper.setProps({ displayUserProfile: undefined, summaryBarData: undefined });
+    expect(wrapper.text()).toEqual('');
+  });
+  
+
+  it('updates correctly when props change', () => {
+    wrapper.setProps({ displayUserId: '456' }); 
+  
```

Comment:
I had a question is there a way we can check on this line wether or not was updated with the props correctly?